generate_wrappers.py

```diff
@@ -139,7 +149,559 @@ def __str__(self):
+        else:
+          body += convert_for_return(self.output_args[0], 'ret')
+        return body
+    elif self.mode == 'eager':
+      body = 'let op = TFE_Op("{}")\n  '.format(self.op_def.name)
+      setters = []
+      for arg in self.input_args:
+        setters.append(arg.swift_setter(self.mode))
+      for attr in self.attrs:
+        setters.append(attr.swift_setter(self.mode, self.string_valued))
+      body += '\n  '.join(setters)
+      counts = ['Int({})'.format(arg.swift_count) for arg in self.output_args]
+      if len(self.output_args) == 0:
+        return body + '\n  op.execute()'
+      body += '\n  return op.execute({})'.format(', '.join(counts))
+      return body
+
+    # `mode` was neither "tfop" nor "eager".
+    raise UnableToGenerateCodeError(
+      'Invalid mode "%s" provided (only "tfop", "eager", and "tfop-eager-fallback" are supported).'
+      % self.mode)
+
+
+class Argument(object):
+  def __init__(self, arg_def, op):
+    self.arg_def = arg_def
+    self.op = op
+    self.is_list = arg_def.number_attr is not '' \
+                   or arg_def.type_list_attr is not ''
+
+  @property
+  def name(self):
+    return self.arg_def.name
+
+  @property
+  def swift_name(self):
+    return swift_compatible_identifier(
+      self.name[0].lower() + self.name[1:])
+
+  @property
+  def swift_arg_name(self):
+    name = self.swift_name
+    if name in _OMITTED_PARAMETER_NAMES:
+      name = '_ ' + name
+    return name
+
+  def swift_type(self, string_valued=False):
+    if string_valued and self.allows_string:
+      return 'StringTensor'
+    return self.type.swift_type
+
+  def swift_handle_type(self, string_valued=False):
+    if string_valued and self.allows_string:
+      return 'TensorHandle<String>'
+    return self.type.swift_handle_type
+
+  def swift_setter(self, mode):
+    if mode == 'tfop':
+      return self.swift_name
+    elif mode == 'eager':
+      number_attr = self.arg_def.number_attr
+      if self.arg_def.number_attr and number_attr not in self.op.inferred_counts:
+        self.op.inferred_counts[number_attr] = self.swift_name + 'Count'
+        return ('let {name}Count = op.addInputList({name})\n  ' +
+                'op.setAttr("{number_attr}", {name}Count)'
+                ).format(name=self.swift_name, number_attr=self.arg_def.number_attr)
+      elif self.is_list:
+        return 'let _ = op.addInputList({})'.format(self.swift_name)
+      else:
+        return 'let _ = op.addInput({})'.format(self.swift_name)
+
+    # `mode` was neither "tfop" nor "eager".
+    raise UnableToGenerateCodeError(
+      'Invalid mode "%s" provided (only "tfop" and "eager" are supported).'
+      % mode)
+
+  @property
+  def swift_count(self):
+    number_attr = self.arg_def.number_attr
+    if number_attr and number_attr in self.op.inferred_counts:
+      return self.op.inferred_counts[number_attr]
+    if number_attr:
+      return self.swift_name + 'Count'
+    if self.arg_def.type_list_attr:
```

Comment:
Can't comment on the raw ops, but these appear to be codegenning as #tfops. This is breaking "saveV2" and "restoreV2" because they no longer return anything. I think the original logic was bad here. They should probably return [AnyTensor] or be otherwise blacklisted.

Reply:
It depends on the `mode` you use. I currently set it to `tfop-eager-fallback` just for backwards compatibility, but the signature should be the same with `eager` mode. I don't understand though what is wrong with the following two ops. `saveV2` does not return anything, as expected, and `restoreV2` returns a value with type `Dtypes` that conforms to `TensorGroup`. This allows you to save and restore say a struct of tensors and avoids the loss of type information incurred by using `[AnyTensor]`. Maybe I am missing something though.
```swift
@inlinable @inline(__always)
public static func saveV2<Dtypes: TensorArrayProtocol>(
  prefix: StringTensor,
  tensorNames: StringTensor,
  shapeAndSlices: StringTensor,
  tensors: Dtypes
) {
  return #tfop("SaveV2",
    prefix,
    tensorNames,
    shapeAndSlices,
    tensors,
    dtypes$dtype: tensors._typeList)
}

@inlinable @inline(__always)
public static func restoreV2<Dtypes: TensorGroup>(
  prefix: StringTensor,
  tensorNames: StringTensor,
  shapeAndSlices: StringTensor
) -> Dtypes {
  let op = TFE_Op("RestoreV2")
  let _ = op.addInput(prefix)
  let _ = op.addInput(tensorNames)
  let _ = op.addInput(shapeAndSlices)
  op.setAttr("dtypes", Dtypes._typeList)
  return op.execute(Int(Dtypes._typeList.count))
}
```

Reply:
My bad, it was just restoreV2 that I had a problem with. You're constraining _typeList to be a static value. This is not useful in the plan that I have. I'll fix it later I guess, but I would prefer disabling it.

Reply:
Actually that was a debate I had. I ended up with somewhat of a middle ground where if the type whose `_typeList` property we want appears as an output arg only, we constrain it to be a `TensorGroup` and use a static property. Otherwise, we constrain it to be a `TensorArrayProtocol` and use an instance property. It’s just that if it’s an output arg in either case we’d need to unpack the tensor handles and that’s something that `TensorGroup` allows us to do. We could disable it for now but I’m curious what use case it doesn’t work for so we can try and think of a better way to do it.

Reply:
I want to determine the type list at runtime. In this case, I will be serializing and deserializing a dynamic list of tensors.

Reply:
That sounds like a special use case that's not easy to generalize over the raw ops generation. Given that the current generation script generates somewhat type-safe code, how about we add an untyped overload for `restoreV2` which offers the functionality you need?